docker/elasticsearch-setup/create-indices.sh

```diff
@@ -5,71 +5,96 @@ set -e
+  RESOURCE_DEFINITION_NAME="$2"
+
+  # query ES to see if the resource already exists
+  RESOURCE_STATUS=$(curl -o /dev/null -s -w "%{http_code}\n" --header "$ELASTICSEARCH_AUTH_HEADER" "$ELASTICSEARCH_URL/$RESOURCE_ADDRESS")
+
+  if [ $RESOURCE_STATUS -eq 200 ]; then
+    # resource already exists -> nothing to do
+    echo -e "\n>>> $RESOURCE_ADDRESS already exists ✓"
+
+  elif [ $RESOURCE_STATUS -eq 404 ]; then
+    # resource doesn't exist -> need to create it
+    echo -e "\n>>> creating $RESOURCE_ADDRESS ..."
+    # use the given path as definition, but first replace all occurences of PREFIX with the actual prefix
+    TMP_SOURCE_PATH="/tmp/$RESOURCE_DEFINITION_NAME"
+    sed -e "s/PREFIX/$PREFIX/g" "$INDEX_DEFINITIONS_ROOT/$RESOURCE_DEFINITION_NAME" | tee -a "$TMP_SOURCE_PATH"
+    curl -s -XPUT --header "$ELASTICSEARCH_AUTH_HEADER" "$ELASTICSEARCH_URL/$RESOURCE_ADDRESS" -H 'Content-Type: application/json' --data "@$TMP_SOURCE_PATH"
+
+  elif [ $RESOURCE_STATUS -eq 401 ] || [ $RESOURCE_STATUS -eq 405 ]; then
+    echo -e "\n>>> failed to GET $RESOURCE_ADDRESS ($RESOURCE_STATUS) !"
+    echo "... make sure you have correct USE_AWS_ELASTICSEARCH env value set (current=$USE_AWS_ELASTICSEARCH)"
+    exit 1
 
-  if [ $(curl -o /dev/null -s -w "%{http_code}" --header "$ELASTICSEARCH_AUTH_HEADER" "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST:$ELASTICSEARCH_PORT/_ilm/policy/${PREFIX}datahub_usage_event_policy") -eq 404 ]
-  then
-    echo -e "\ncreating datahub_usage_event_policy"
-    sed -e "s/PREFIX/${PREFIX}/g" /index/usage-event/policy.json | tee -a /tmp/policy.json
-    curl -XPUT --header "$ELASTICSEARCH_AUTH_HEADER" "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST:$ELASTICSEARCH_PORT/_ilm/policy/${PREFIX}datahub_usage_event_policy" -H 'Content-Type: application/json' --data @/tmp/policy.json
-  else
-    echo -e "\ndatahub_usage_event_policy exists"
-  fi
-  if [ $(curl -o /dev/null -s -w "%{http_code}" --header "$ELASTICSEARCH_AUTH_HEADER" "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST:$ELASTICSEARCH_PORT/_index_template/${PREFIX}datahub_usage_event_index_template") -eq 404 ]
-  then
-    echo -e "\ncreating datahub_usage_event_index_template"
-    sed -e "s/PREFIX/${PREFIX}/g" /index/usage-event/index_template.json | tee -a /tmp/index_template.json
-    curl -XPUT --header "$ELASTICSEARCH_AUTH_HEADER" "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST:$ELASTICSEARCH_PORT/_index_template/${PREFIX}datahub_usage_event_index_template" -H 'Content-Type: application/json' --data @/tmp/index_template.json
   else
-    echo -e "\ndatahub_usage_event_index_template exists"
+    echo -e "\n>>> unexpected $RESOURCE_ADDRESS status $RESOURCE_STATUS !"
+    exit 1
   fi
 }
 
+# create indices for ES (non-AWS)
+function create_datahub_usage_event_datastream() {
+  create_if_not_exists "_ilm/policy/${PREFIX}datahub_usage_event_policy" policy.json
+  create_if_not_exists "_index_template/${PREFIX}datahub_usage_event_index_template" index_template.json
```

Comment:
Do we not need a line like this here:

```
 create_if_not_exists "${PREFIX}datahub_usage_event" aws_es_usage_event.json
```

to ensure that the index is actually created for non aws cases?

Reply:
Because I do think there's a separate issue actively open where on fresh quickstarts without any usage events, the analytics index will be missing

Reply:
I'm not sure how this works outside of AWS. The behavior in non-AWS environment should be the same as before refactoring.

Can you link the issue mentioned?

Reply:
It's this PR in question: https://github.com/datahub-project/datahub/pull/5974